infra/modules/providers/azure/storage-account/README.md

```diff
@@ -0,0 +1,118 @@
+
+- Ability to deploy a Storage Container within the Storage Account.
+
+## Definition
+
+Azure Storage definition example:
+
+```terraform
+resource "azurerm_storage_account" "sa" {
+  resource_group_name      = var.resource_group_name
+  location                 = var.resource_group_location
+  name                     = lower(var.account_name)
+  account_tier             = var.performance_tier
+  ...
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_storage_container" "sa" {
+  name                  = var.storage_container_name
+  resource_group_name   = var.resource_group_name
+  storage_account_name  = azurerm_storage_account.sa.name
+}
+
+module "service-principal" {
+  source       = "../service-principal"
+  object_id    = var.existing_sp_object_id
+  role_name    = var.storage_role_definition_name
+  role_scope   = azurerm_storage_account.sa.id
+}
+```
+
+Terraform resources directly referenced within the `storage-account` module include the following:
+
+- [azurerm_storage_account](https://www.terraform.io/docs/providers/azurerm/r/storage_account.html)
+
+- [azurerm_storage_container](https://www.terraform.io/docs/providers/azurerm/r/storage_container.html)
+
+## Usage
+
+App Monitoring usage example:
+
+```terraform
+resource "azurerm_resource_group" "main" {
+  name     = var.prefix
+  location = var.resource_group_location
+}
+
+module "service_plan" {
+  source              = "../../modules/providers/azure/service-plan"
+  resource_group_name = azurerm_resource_group.main.name
+  service_plan_name   = "${azurerm_resource_group.main.name}-sp"
+}
+
+module "app_service" {
+  source                           = "../../modules/providers/azure/app-service"
+  app_service_name                 = var.app_service_name
+  service_plan_name                = module.service_plan.service_plan_name
+  ...
+}
+
+module "storage_account" {
+  source                            = "../../modules/providers/azure/storage-account"
+  resource_group_name               = "${azurerm_resource_group.main.name}"
+  resource_group_location           = "${azurerm_resource_group.main.location}"
+  account_name                      = "teststorageaccount"
+  storage_container_name            = "test-storage-container"
+  encryption_source                 = "Microsoft.Storage"
+  existing_sp_object_id             = "${module.app_service.app_service_identity_object_ids[0]}"
+}
+```
+
+#### Configuring `storage account`
+
+Please visit _**Azure Storage's**_ [Introduction to Azure Storage](https://docs.microsoft.com/en-us/azure/storage/common/storage-introduction.) page for more information on configuring a storage account.
+
+#### Configuring `rbac role`
+
+Please visit _**Cobalt's**_ [service principal module](../service-principal) page for more information on configuring a service principal and rbac role.
+
+## Argument Reference
+
+Supported arguments for this module are available in [variables.tf](variables.tf).
```

Comment:
IMO this would be great to call out in the docs.

Reply:
I'm more in favor of keeping the docs very light-weight. I think there's more value in incentivizing a reader to become familiar with the variables.tf as it's a very straight forward terraform file.